### PR TITLE
BCP 47 language tags

### PR DESCRIPTION
**Pull request by ydnar**
This pull request implements a new field on each zone: `languages`, containing a list of one or more canonical BCP 47 language tags, with the language code, optional script, and country/region modifier, e.g.:

```json
{
	"domain": "中国",
	"languages": [
		"zh-Hans-CN"
	]
}
```

```json
{
	"domain": "укр",
	"languages": [
		"crh-UA",
		"mul-Cyrl-UA",
		"ru-UA",
		"uk-UA"
	]
}
```

### CODE REVIEW

**Comment by case** [comment]
Awesome, thanks @ydnar! Could you add an example command to the readme? (e.g. to add a language to a zone)

**Comment by ydnar** [comment]
@case @cee-dub @chriskillpack this is ready for review. It has a bit of incidental work unrelated to the core language tags (some metadata cleanup), but they touch the same zones. If you approve this, we can iterate in future PRs to add language metadata to the other IDN zones.

File: zone.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ type Zone struct {
 	// Locations associated with the Zone
 	Locations []string
 
+	// BCP 47 language tag associated with this Zone
```
Comment (cee-dub) [inline zone.go:53]: Tags, right?

File: zone_test.go [hunk 2]
```diff
@@ -17,3 +18,28 @@ func TestSizeofZone(t *testing.T) {
...
+func TestLanguages(t *testing.T) {
+	tests := []struct {
+		domain string
+		want   []string
+	}{
+		{"com", nil},
+		{"中国", []string{"zh-Hans-CN"}},
+		{"中國", []string{"zh-Hant-CN"}},
+		{"台灣", []string{"zh-Hant-TW"}},
+		{"台灣", []string{"zh-Hant-TW"}},
```
Comment (cee-dub) [inline zone_test.go:31]: This appears to be identical to the one above

**Review by cee-dub — APPROVED** [review]
Looks like a couple tiny things to fix and also the merge conflict.

File: cmd/zonedb/main.go [hunk 3]
```diff
@@ -39,6 +39,8 @@ func main() {
 	checkPS := flag.Bool("ps", false, "check against Public Suffix List")
 
 	// Mutate operations
+	addLanguages := flag.String("add-languages", "", "add BCP 48 language tags to zones (comma-delimited)")
```
[the remark this answers is not in the document]
Comment (ydnar) [reply cmd/zonedb/main.go:42]: Good catch

File: zone_test.go (hunk 2 above)
Comment (ydnar) [reply zone_test.go:31]: Fixed, thanks!

File: zone.go (hunk 1 above)
Comment (ydnar) [reply zone.go]: fixed

**Pull request merged**